src/main/java/bio/terra/common/iam/TokenAuthenticatedRequestFactory.java

```diff
@@ -0,0 +1,28 @@
+package bio.terra.common.iam;
+
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.stereotype.Component;
+
+/**
+ * Implementation of {@link AuthenticatedUserRequestFactory} when HTTP requests contain a bearer
+ * token in the Authorization header.
+ *
+ * <p>This factory is all that should be required to propagate the caller's identity in most cases.
+ * In cases where Sam is not otherwise called to check the caller's access or the caller's email or
+ * id are required, seel{@link SamAuthenticatedUserRequestFactory}.
+ */
+@Component
+public class TokenAuthenticatedRequestFactory {
+
+  static final String AUTHORIZATION = "Authorization";
+
+  public TokenAuthenticatedRequest from(HttpServletRequest servletRequest) {
+    TokenAuthenticatedRequest.Builder builder = TokenAuthenticatedRequest.builder();
+
+    String authHeader = servletRequest.getHeader(AUTHORIZATION);
```

Comment:
Technically we also get token expiration from the Apache proxy, but I don't know of any services which actually use it -- so fine to leave it out I think.

Reply:
I don't think expiration is ever used. The preexisting code did not get it either.

Reply:
Is there a way to get the expiration after the fact if we need it?

Reply:
it is in the header if we need it. In all the years of FireCloud's and Terra's existence it has been there and we have never used it.

Reply:
SGTM